### PR TITLE
[Feature] Custom action parameters

### PR DESCRIPTION
**Pull request by mproch**
Peek 2021-03-24 20-07

### CODE REVIEW

File: ui/client/components/modals/CustomActionDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+  const processId = useSelector(getProcessId)
+  const dispatch = useDispatch()
+  const action = useSelector(getModalDialog).customAction
+
+  const empty = (action?.parameters || []).reduce((obj, param) => ({...obj, [param.name]: ""}), {})
+
+  const [mapState, setState] = useState(empty)
+
+  const init = useCallback(() => setState(empty), [])
```
Comment (JulianWielga) [inline ui/client/components/modals/CustomActionDialog.tsx:22]: https://reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect

File: ui/client/components/modals/CustomActionDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+  const confirm = useCallback(async () => {
+    await dispatch(HttpService
+      .customAction(processId, action.name, mapState)
+      .finally(() => dispatch(loadProcessState(processId))))
+  }, [processId, action, mapState])
+
+  const renderFieldLabel = (label) => {
```
Comment (JulianWielga) [inline ui/client/components/modals/CustomActionDialog.tsx:30]: I think it gives nothing positive to have this function, its cleaner to have this div in place or extract it as component.

Comment (mproch) [reply ui/client/components/modals/CustomActionDialog.tsx:30]: :+1: true, it was copied from elswhere...

File: ui/client/components/modals/CustomActionDialog.tsx (hunk 1 above)
Comment (mproch) [reply ui/client/components/modals/CustomActionDialog.tsx]: I added comment...

**Review by JulianWielga — APPROVED** [review]

**Pull request merged**